src/APL/vercel-kv/vercel-kv-apl.ts

```diff
@@ -29,44 +34,133 @@ export class VercelKvApl implements APL {
   async get(saleorApiUrl: string): Promise<AuthData | undefined> {
     this.debug("Will call Vercel KV to get auth data for %s", saleorApiUrl);
 
-    try {
-      const authData = await kv.hget<AuthData>(this.hashCollectionKey, saleorApiUrl);
-
-      return authData ?? undefined;
-    } catch (e) {
-      this.debug("Failed to get auth data from Vercel KV");
-      this.debug(e);
-
-      throw e;
-    }
+    return this.tracer.startActiveSpan(
+      "VercelKvApl.get",
+      {
+        attributes: {
+          saleorApiUrl,
+          [SemanticAttributes.PEER_SERVICE]: OTEL_APL_SERVICE_NAME,
+          [SemanticAttributes.HTTP_METHOD]: "GET",
+        },
+        kind: SpanKind.CLIENT,
+      },
+      async (span) => {
+        try {
+          const authData = await kv.hget<AuthData>(this.hashCollectionKey, saleorApiUrl);
+
+          this.debug("Received response from VercelKV");
+
+          if (!authData) {
+            this.debug("AuthData is empty for %s", saleorApiUrl);
+          }
+
+          span
+            .setStatus({
+              code: 200,
+              message: "Received response from VercelKV",
+            })
+            .end();
+
+          return authData ?? undefined;
+        } catch (e) {
+          this.debug("Failed to get auth data from Vercel KV");
+          this.debug(e);
+
+          span.recordException("Failed to get auth data from Vercel KV");
```

Comment:
question: how is this different than `span.setStatus`?

Reply:
AFAIK:
- setStatus marks span to be error, its long-term operation with a result
- recordException is a single point of time, like log. It sets precise event on span, that it happened just there. 

I'm not sure if this is correctly displayed in Datadog, but since we are still exploring this, I'd rather leave it and have more information. App-sdk itself doesn't send otel logs, so this could be a way to have them